pkg/subnet/publisher.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnet
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	git "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+type Publisher interface {
+	Publish(r *git.Repository, subnetName, vmName string, subnetYAML []byte, vmYAML []byte) error
+	GetRepo() (*git.Repository, error)
+}
+
+type publisherImpl struct {
+	alias    string
+	repoURL  string
+	repoPath string
+}
+
+var _ Publisher = &publisherImpl{}
+
+func NewPublisher(repoDir, repoURL, alias string) Publisher {
+	repoPath := filepath.Join(repoDir, alias)
+	return &publisherImpl{
+		alias:    alias,
+		repoURL:  repoURL,
+		repoPath: repoPath,
+	}
+}
+
+func (p *publisherImpl) GetRepo() (repo *git.Repository, err error) {
+	// path exists
+	if _, err = os.Stat(p.repoPath); err == nil {
```

Comment:
for err pattern, you probably want to use `:=` instead of `=`, and keep that variable in the if score, as you don't need and probably don't want to modify a previous err value

Reply:
applies to other occurrences of the end pattern

Reply:
Agreed. Although it can always be assumed that the previous err value has been checked (the linter will complain if not).